library/src/webapp/editor/ckextraplugins/common/js/itemsearch.js

```diff
@@ -91,28 +106,41 @@ $.fn.itemSearch = function(options) {
     var pagesContainer = $('<div/>').addClass('pages');
 
     // build page numbers and containers
-    for (i = 1; i <= pages; i++) {
-      nav.append($('<a/>').html(i).attr({'class': 'pageNum', 'href': '#', 'data-page': i}));
-      pagesContainer.append($('<div/>').attr({'class': 'page', 'data-page': i}));
-    };
 
+    var currentPage = $.fn.itemSearch.currentPage;
+    var iFrameId = settings.resultsContainer[0].id.replace('Results', 'Iframe');
+    if (currentPage >= 1 && currentPage  <=4){
+      if (currentPage  > 1){
+          appendBackButton(nav, currentPage, iFrameId );
+      }
+      for (i = 1; i <= 7; i++) {
+          appendPageButtons(i, currentPage , pagesContainer, nav, iFrameId);
+      }
+    }
+    else {
+        appendBackButton(nav, currentPage );
+        for (i = currentPage-3; i <= currentPage +3; i++) {
+            appendPageButtons(i, currentPage , pagesContainer, nav, iFrameId);
+        }
+    }
+
+    nav.append($('<a/>').html('<span class="pageSpan">Next »</span>').attr({'class': 'pageNum ' + iFrameId, 'href': '#', 'data-page': currentPage +1}));
     container.append(pagesContainer);
 
     // now move results into the correct containers
     for (i = 0; i < results.length; i++) {
-      var page = Math.ceil((i+1) / settings.pagination);
-      container.find('.page[data-page="' + page + '"]').append(results[i]);
+      container.find('.page[data-page="' + currentPage + '"]').append(results[i]);
     }
 
     $(document).on('click', '.pageNum', function() {
       // find closest results container
```

Comment:
I've removed references to the localContainer in this method which was controlling the old js way of 'paging' the results.  I should have also removed this (now) redundant comment (or reworded it really)-  which I'll do.